Kitodo/src/main/java/org/kitodo/production/model/Subfolder.java

```diff
@@ -303,6 +306,31 @@ public Optional<URI> getURIIfExists(String canonical) {
         return new File(uri.getPath()).isFile() ? Optional.of(uri) : Optional.empty();
     }
 
+    /**
+     * Checks if the folder determined by determineDirectoryAndFileNamePattern() is empty.
+     * Retrieve the directory URI and lists its contents.
+     * It stops checking as soon as the first file is found.
+     *
+     * @return true if the folder is empty, false if it contains at least one file
+     */
+    public boolean isFolderEmpty() {
+        Pair<URI, Pattern> query = determineDirectoryAndFileNamePattern();
+        File dir = fileService.getFile(query.getLeft());
+        if (!dir.exists() || !dir.isDirectory()) {
+            return false;  // Ensures we only check valid directories
+        }
+        try (Stream<Path> entries = Files.list(dir.toPath())) {
+            // Apply regex filtering to check only for files matching the pattern
+            return !entries
+                    .map(Path::getFileName)
+                    .map(Path::toString)
+                    .anyMatch(name -> query.getRight().matcher(name).matches()); // Stop after first match
+        } catch (IOException e) {
+            return false;
```

Comment:
Should an rised IOException interpreted as and directory with content? I'm unsure about this but it could be okay. It could be helpful if this exception got logged in trace or debug log level.

Reply:
IOException now marks the folder has having no images.